src/main/java/com/google/api/codegen/discovery/config/go/GoTypeNameGenerator.java

```diff
@@ -0,0 +1,71 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery.config.go;
+
+import java.util.List;
+
+import com.google.api.codegen.discovery.DefaultString;
+import com.google.api.codegen.discovery.config.TypeNameGenerator;
+import com.google.api.codegen.util.Name;
+import com.google.common.base.Strings;
+
+public class GoTypeNameGenerator implements TypeNameGenerator {
```

Comment:
Yeah, I noticed that as well... the scope of this class expanded a bit since it was first conceptualized.

I think the right move is just to rename it - it's a utility for generating language-specific content in the SampleConfig.

I've made note of it and will refactor once the rate of MVVM changes settles down.